IBaseEntity/IBaseEntity.h

```diff
@@ -1,17 +1,27 @@
 #pragma once
 
-#include "IBaseEntity_flags.h"
 #include "System/Globals.h"
 #include "System/Input.h"
 
+#define BF(i)             (1UL << i)
+#define FL_NODAMAGE       BF(0) // ignores all damage whatsoever
+#define FL_INVINCIBLE     BF(1) // damage events happen, but never actually dies
+#define FL_ROUND_PRESERVE BF(2) // round restarts don't respawn this entity
+#define FL_ROUND_DESTROY  BF(3) // round restarts destroy this entity
+#define FL_IGNORE_USE     BF(4) // entity ignores generic "+Use" functionality
+#define FL_INGORE_INPUT                                                             \
+    BF(5) // entity ignores all player controller input whatsoever
+#define FL_EXTRA   BF(6)
+#define FL_EXTRA_2 BF(7)
+
 /**
  *-------------------------------------------------------------------------------------
  * IBaseEntity
  * IBaseEntity is an interface which declares functionality common to all "Base"
  *types and defines some of the most basic functions.
  *-------------------------------------------------------------------------------------
  */
-INTERFACE IBaseEntity {
+interface IBaseEntity {

```

Comment:
Renamed `INTERFACE` to `interface`. `interface` is not a keyword in C++, but this works just as well. If `interface` is ever added to C++, we don't need to change anything